Source/EasyNetQ/Scheduling/DeadLetterExchangeAndMessageTtlScheduler.cs

```diff
@@ -8,66 +8,76 @@ namespace EasyNetQ.Scheduling
 {
     public class DeadLetterExchangeAndMessageTtlScheduler : IScheduler
     {
+        private readonly ConnectionConfiguration configuration;
         private readonly IAdvancedBus advancedBus;
         private readonly IConventions conventions;
         private readonly IExchangeDeclareStrategy exchangeDeclareStrategy;
         private readonly IMessageDeliveryModeStrategy messageDeliveryModeStrategy;
 
         public DeadLetterExchangeAndMessageTtlScheduler(
+            ConnectionConfiguration configuration,
             IAdvancedBus advancedBus,
             IConventions conventions,
             IMessageDeliveryModeStrategy messageDeliveryModeStrategy,
             IExchangeDeclareStrategy exchangeDeclareStrategy
         )
         {
+            Preconditions.CheckNotNull(configuration, "configuration");
             Preconditions.CheckNotNull(advancedBus, "advancedBus");
             Preconditions.CheckNotNull(conventions, "conventions");
             Preconditions.CheckNotNull(messageDeliveryModeStrategy, "messageDeliveryModeStrategy");
             Preconditions.CheckNotNull(exchangeDeclareStrategy, "exchangeDeclareStrategy");
 
+            this.configuration = configuration;
             this.advancedBus = advancedBus;
             this.conventions = conventions;
             this.messageDeliveryModeStrategy = messageDeliveryModeStrategy;
             this.exchangeDeclareStrategy = exchangeDeclareStrategy;
         }
 
-        public async Task FuturePublishAsync<T>(T message, TimeSpan delay, string topic = "",
-            CancellationToken cancellationToken = default)
+        /// <inheritdoc />
+        public async Task FuturePublishAsync<T>(
+            T message, TimeSpan delay, string topic, CancellationToken cancellationToken = default
+        )
         {
             Preconditions.CheckNotNull(message, "message");
             Preconditions.CheckNotNull(topic, "topic");
 
+            using var cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+            if (configuration.Timeout != Timeout.InfiniteTimeSpan)
+                cts.CancelAfter(configuration.Timeout);
```

Comment:
This could be extracted to extension method like
```c#
public static CancellationTokenSource Limit(this CancellationToken token, TimeSpan timeout)
{
    using var cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
            if (timeout != Timeout.InfiniteTimeSpan)
                cts.CancelAfter(timeout);
    return cts;
}
```